### PR TITLE
Replace ScopedPtr with unique_ptr

### PR DESCRIPTION
**Pull request by turtleli**
Since we're now using compilers with decent C++11 support, we can use unique_ptr instead of the ScopedPtr.

This leaves ScopedPtrMT as the only remaining custom smart pointer class.

### CODE REVIEW

**Comment by gregory38** [comment]
I really think it would be better to put <memory> in a global header.

**Comment by turtleli** [comment]
It's been added to common/include/Utilities/Dependencies.h, which seems to be used by everything.

**Comment by gregory38** [comment]
```
This leaves ScopedPtrMT as the only remaining custom smart pointer class.
```
Hum, could it be something similar to shared ptr ? (dunno what the code do)

**Comment by turtleli** [comment]
It's similar to ScopedPtr, but I think it's meant to be thread safe.

File: pcsx2/gui/App.h [hunk 1]
```diff
@@ -241,12 +242,12 @@ class pxAppResources
...
-	ScopedPtr<wxBitmap>			ScreenshotBitmap;
-	ScopedPtr<AppGameDatabase>	GameDB;
+	std::unique_ptr<wxImageList>		ConfigImages;
+	std::unique_ptr<wxImageList>		ToolbarImages;
+	std::unique_ptr<wxIconBundle>		IconBundle;
+	std::unique_ptr<wxBitmap>			Bitmap_Logo;
+	std::unique_ptr<wxBitmap>			ScreenshotBitmap;
+	std::unique_ptr<AppGameDatabase>	GameDB;
 
 	pxAppResources();
```
Comment (ssakash) [inline pcsx2/gui/App.h]: ``ImageId`` , ``Bitmap_Logo`` and ``ScreenshotBitmap`` aren't properly aligned.

File: pcsx2/gui/Panels/ConfigurationPanels.h [hunk 2]
```diff
@@ -620,8 +621,8 @@ namespace Panels
 		ComboBoxPanel*	m_ComponentBoxes;
 		bool			m_Canceled;
 
-		ScopedPtr<wxArrayString>	m_FileList;	// list of potential plugin files
-		ScopedPtr<EnumThread>		m_EnumeratorThread;
+		std::unique_ptr<wxArrayString>	m_FileList;	// list of potential plugin files
+		std::unique_ptr<EnumThread>		m_EnumeratorThread;
```
Comment (ssakash) [inline pcsx2/gui/Panels/ConfigurationPanels.h]: ``m_EnumeratorThread`` is not aligned here.

File: pcsx2/x86/microVU.h [hunk 3]
```diff
@@ -192,10 +193,10 @@ struct microVU {
...
 	u32 cacheSize;		// VU Cache Size
 
-	microProgManager			prog;		// Micro Program Data
-	microProfiler               profiler;   // Opcode Profiler
-	ScopedPtr<microRegAlloc>	regAlloc;	// Reg Alloc Class
-	ScopedPtr<AsciiFile>		logFile;	// Log File Pointer
+	microProgManager				prog;		// Micro Program Data
+	microProfiler					profiler;   // Opcode Profiler
+	std::unique_ptr<microRegAlloc>	regAlloc;	// Reg Alloc Class
+	std::unique_ptr<AsciiFile>		logFile;	// Log File Pointer
```
Comment (ssakash) [inline pcsx2/x86/microVU.h]: needs alignment here.

**Comment by gregory38** [comment]
```
It's been added to common/include/Utilities/Dependencies.h, which seems to be used by everything.
```
Ok. Does it still need to be included in the remaining cpp?

```
It's similar to ScopedPtr, but I think it's meant to be thread safe.
```
Ok. So we can keep it :)

File: pcsx2/gui/App.h (hunk 1 above)
Comment (turtleli) [inline pcsx2/gui/App.h]: It's aligned if tabs are 4 spaces wide.

**Comment by turtleli** [comment]
> Ok. Does it still need to be included in the remaining cpp?

Probably yes for some files? Headers are a mess!

>> It's similar to ScopedPtr, but I think it's meant to be thread safe.

> Ok. So we can keep it :)

I think so, it's only used in one place - the pxThread stuff..

File: pcsx2/gui/App.h (hunk 1 above)
Comment (ssakash) [inline pcsx2/gui/App.h]: yeah, it seems to be fine when indent size is set to 4. (I guess VS also uses 4 ? ) 

Is there any way to have 4 as the default value for all the files which are viewed through at Git ?

Comment (turtleli) [inline pcsx2/gui/App.h]: 4 is the default VS setting (I think).

Dunno for your other question. Someone else will probably know/find out ;)

Comment (ssakash) [inline pcsx2/gui/App.h]: FWIW I just forced the tab indent size on user styles addon and it seems to be properly aligned now :) 

pic

**Comment by gregory38** [comment]
Well it isn't critical. Honestly the next thing that would worth to be ported is atomic/locking stuff. However we must keep thread (pthread). Tests done on GSdx weren't conclusive.

**Comment by gregory38** [comment]
@ssakash 
It seems Github add support for editorconfig ( http://editorconfig.org/ ). It would be very nice to have something for the project.

**Comment by gregory38** [comment]
@turtleli 
Instead to report all of them in the diff :) If it up to you to use .reset or the affectation operator `=`.

**Comment by ssakash** [comment]
> It seems Github add support for editorconfig ( http://editorconfig.org/ ). It would be very nice to have something for the project.

Hmm, that looks interesting. but it seems to require an additional plugin for other IDE's, the ``trim_trailing_whitespace`` option would help me a lot to prevent nagging from turtleli :P 

by the way, does it also work on the Git online editor ? (I'd guess it does). I'll try playing with it.

**Comment by gregory38** [comment]
Well it isn't clear for me what you can do with it. I will be nice if it can work with the diff viewer of github.

@turtleli I'm fine with your change. However I want to test the exception stuff (ScopedExcept), to be sure Exception are properly routed to the console.

**Comment by turtleli** [comment]
@gregory38 
> If it up to you to use .reset or the affectation operator =.

I'll leave it as the `=`. Originally I wanted to use `make_unique` for most of it, but I don't think GCC 4.8 supports it and the compile flags would need changed too.

I could probably split a few of the lines - I originally did so, but thought it would be slightly easier to review if I didn't do much reformatting so I changed it back.

> I'm fine with your change. However I want to test the exception stuff (ScopedExcept), to be sure Exception are properly routed to the console.

No problem.

@ssakash 

> the trim_trailing_whitespace option would help me a lot to prevent **nagging** from turtleli :P

No worries, I'll find other stuff about your code to nag about. ;)

**Pull request merged**